plugins/module_utils/managed_cluster_addons/application_manager.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.ocmplus.cm.plugins.module_utils.addon_utils import (
+    check_cluster_management_addon_feature,
+    ensure_klusterlet_addon,
+)
+
+
+class application_manager:
+    def __init__(self, module: AnsibleModule, enabled, hub_client, managed_cluster_name, addon_name, wait=False, timeout=60):
+        self.module = module
+        self.enabled = enabled
+        self.hub_client = hub_client
+        self.managed_cluster_name = managed_cluster_name
+        self.addon_name = addon_name
+        self.wait = wait
+        self.timeout = timeout
+
+    def run(self):
```

Comment:
so each of the class will implement a `run()` function? I feel this naming is not very clear for what it will do (as we may add other behaviour in future)

Reply:
Create three functions now:
`check_feature()`
`enable_addon()`
`disable_addon()`